tortoisebeacon/tortoisebeacon.go

```diff
@@ -588,3 +530,22 @@ func (tb *TortoiseBeacon) votingThreshold() int {
 func (tb *TortoiseBeacon) atxThresholdFraction(totalWeight int) float64 {
 	return 1 - math.Pow(2.0, -(float64(tb.config.Kappa)/((1.0-tb.config.Q)*float64(totalWeight))))
 }
+
+// TODO(nkryuchkov): add unit tests
+func (tb *TortoiseBeacon) beaconAlreadyCalculated(layerID types.LayerID) bool {
```

Comment:
rename - this function tells us if new beacon should be ready right? it doesn't guarantee it's ready

Reply:
Yes, it does this. Should it also check for a beacon value explicitly?